skyrim-platform/src/platform_se/codegen/TSConverter.cpp

```diff
@@ -0,0 +1,251 @@
+
+#include "nlohmann/json.hpp"
+#include <algorithm>  // For transform
+#include <cctype>     // for toupper() and tolower()
+#include <filesystem> // For exist()
+#include <fstream>    // For ifstream and ofstream
+#include <functional> // For std::function
+#include <iomanip>    // For setw()
+#include <iostream>
+#include <iostream> // For stringstream
+#include <set>
+
+int main(int argc, char* argv[])
+{
+  const std::filesystem::path pathToJsonFile = argv[1];
+  const std::filesystem::path pathToPapyrusClassesFile = argv[2];
+  const std::filesystem::path pathToTypeScriptFile = argv[3];
+
+  if (!std::filesystem::exists(pathToJsonFile)) {
+    std::cout << "Json file: " << pathToJsonFile << " dosn't exits, check it.";
+    return 1;
+  }
+  if (!std::filesystem::exists(pathToPapyrusClassesFile)) {
+    std::cout << "Default papyrus classes file: " << pathToPapyrusClassesFile
+              << " dosn't exits, check it.";
+    return 2;
+  }
+
+  std::ifstream input(pathToJsonFile);
+  std::ifstream papyrusClasses(pathToPapyrusClassesFile);
+  std::ofstream output(pathToTypeScriptFile);
+
+  enum class mode
+  {
+    module = 0,
```

Comment:
Enum names must start with an upper case: `enum SomeEnum`. This rule is also applied to enum constants.

Also, we should not use `module` as a constant name. It may break IDE.